app-example/src/main/java/com/badoo/ribs/example/app_bar/AppBarView.kt

```diff
@@ -3,12 +3,16 @@ package com.badoo.ribs.example.app_bar
 import android.view.ViewGroup
 import android.widget.ImageView
 import androidx.annotation.LayoutRes
+import androidx.lifecycle.DefaultLifecycleObserver
```

Comment:
Comparing all 3 pull requests in terms of resilience this one seems the best for me.

In the: "View + Lifecycle, RibView approach" we need manually on the framework level invoke `onCreate(lifecycle: Lifecycle)` if we forget to do it for some reason (migration) our `lateinit` approach will be broken on the individual RIB level.

In the: "View + Lifecycle, Plugin approach" we need to check does our View `RibView.LifecycleAware` view which is ok, but a little bit not so nice as approach in current pr.
Seems like we open flexibility for developers to either use/not_use this approach and I am not sure is it ok or not.
And we can also forget to call `onCreate()` as above.

Current approach will force developers to follow same rules when implementing View. However this seems the most difficult in terms of migrations which needs to be considered.

I would range the prs like that:
1. View + Lifecycle, Constructor approach
2. View + Lifecycle, Plugin approach
3. View + Lifecycle, RibView approach

Reply:
I agree.
Out of the 3, this option is my favourite one.
The other two require the client code to override the `onCreate` callback, which can be confusing, and especially when inside of that `onCreate` function, we subscribe to the framework's lifecycle updates...
This solution, as it is using the factory and the view's constructor, makes it much clearer to "opt-in" in lifecycle updates.
My only concern was about the verbosity of the client code (see example below), but as mentioned in this PR, it should be improved with `fun interface` in Kotlin 1.4.
Example: 
```   
override fun invoke(deps: Nothing?): NodeViewFactory<AppBarView> =
            object : NodeViewFactory<AppBarView> {
                override fun invoke(p1: RibView, p2: Lifecycle): AppBarView =
                    AppBarViewImpl(
                        androidView = p1.inflate(layoutRes),
                        lifecycle = p2
                    )
            }
```